tests/phpunit/tests/option/option.php

```diff
@@ -524,4 +524,147 @@ public function data_update_option_type_juggling() {
 			array( null, array(), true ),
 		);
 	}
+
+	/**
+	 * @ticket 22192
+	 *
+	 * @covers ::update_option
+	 *
+	 * @dataProvider data_update_option_type_juggling
+	 *
+	 * @param mixed $old_value One of the values to compare.
+	 * @param mixed $new_value The other value to compare.
+	 */
+	public function test_update_option_should_hit_cache_when_loosely_equal_to_existing_value_and_cached_values_are_faithful_to_original_type( $old_value, $new_value ) {
+		add_option( 'foo', $old_value );
+		$num_queries = get_num_queries();
+
+		$updated = update_option( 'foo', $new_value );
+
+		$this->assertFalse( $updated, 'update_option should not return true when values are loosely equal.' );
+		$this->assertSame( $num_queries, get_num_queries(), 'The number of database queries should not change.' );
+	}
+
+	/**
+	 * @ticket 22192
+	 *
+	 * @covers ::update_option
+	 *
+	 * @dataProvider data_update_option_type_juggling
+	 *
+	 * @param mixed $old_value One of the values to compare.
+	 * @param mixed $new_value The other value to compare.
+	 */
+	public function test_update_option_should_hit_cache_when_loosely_equal_to_existing_value_and_cached_values_are_pulled_from_the_database( $old_value, $new_value ) {
+		add_option( 'foo', $old_value );
+		wp_cache_delete( 'alloptions', 'options' );
+		wp_load_alloptions();
+
+		$num_queries = get_num_queries();
+
+		$updated = update_option( 'foo', $new_value );
+
+		$this->assertSame( (string) $old_value, get_option( 'foo' ), 'The value should not have been changed.' );
```

Comment:
This is problematic causing test failures: We cannot unconditionally parse the `get_option()` result to a string as that doesn't work for arrays.
```suggestion
		$expected_value = is_array( $old_value ) ? $old_value : (string) $old_value;
		$this->assertSame( $expected_value, get_option( 'foo' ), 'The value should not have been changed.' );
```

More importantly though, this test covers the same situation that is already covered in `test_update_loosey_options_from_refreshed_cache` from https://core.trac.wordpress.org/changeset/56648. So we should add the two additional assertions (for the old value, and for the number of database queries) to that test instead of having this duplicate test.